### PR TITLE
Add new fitting executable

### PR DESCRIPTION
**Pull request by ahnitz**
Adds a new executable that allows one to specific the parameters and the smoothing width. A n-dimensional squashed sphere is constructed around each template and currently the average of the fitting values within the sphere is assigned to the template.

### CODE REVIEW

**Comment by ahnitz** [comment]
@tdent This isn't probably ready to review, but I'm opening this so you can see what I was talking about.

File: bin/hdfcoinc/pycbc_fit_sngls_over_multiparam [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+tid = fits['template_id'][:]
+
+logging.info('Calculating template parameter values')
+bank = h5py.File(args.bank_file, 'r')
+m1, m2, s1z, s2z = trstats.get_masses(bank, tid)
+
+parvals = []
+for param, slog in zip(args.fit_param, args.log_param):
+    data = trstats.get_param(param, args, m1, m2, s1z, s2z)
+    if slog == 'false':
```
Comment (tdent) [inline bin/hdfcoinc/pycbc_fit_sngls_over_multiparam]: I suggest to accept both 'false' and 'False' etc., or (better) implement explicit choices in the arg parser

File: bin/hdfcoinc/pycbc_fit_sngls_over_multiparam [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                    help="hdf file containing template parameters. Required "
+                         "unless reading param from template fit file")
+parser.add_argument("--output", required=True,
+                    help="Location for output file containing smoothed fit "
+                         "coefficients.  Required")
+parser.add_argument("--use-template-fit-param", action="store_true",
+                    help="Use parameter values stored in the template fit file"
+                         "as template_param for smoothing.", default=False)
+parser.add_argument("--fit-param", nargs='+',
+                    help="Parameter over which to regress the background "
```
Comment (tdent) [inline bin/hdfcoinc/pycbc_fit_sngls_over_multiparam]: Should rewrite the help here to emphasize that several parameters may be passed and also display the new choices (eta and chi_eff)

File: bin/hdfcoinc/pycbc_fit_sngls_over_multiparam [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                    help="Approximant for template duration. Default SEOBNRv4")
+parser.add_argument("--f-lower", type=float, default=0.,
+                    help="Starting frequency for calculating template "
+                         "duration, if not reading from the template fit file")
+parser.add_argument("--min-duration", type=float, default=0.,
+                    help="Fudge factor for templates with tiny or negative "
+                         "values of template_duration: add to duration values"
+                         " before fitting. Units seconds.")
+parser.add_argument("--log-param", nargs='+',
+                    help="Take the log of the fit param before smoothing.")
```
Comment (tdent) [inline bin/hdfcoinc/pycbc_fit_sngls_over_multiparam]: rewrite the help to say you need a True or False for each param in list order

File: bin/hdfcoinc/pycbc_fit_sngls_over_multiparam [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+parser.add_argument("--smoothing-width", type=float, nargs='+', required=True,
+                    help="Distance in the space of fit param values (or the "
```
Comment (tdent) [inline bin/hdfcoinc/pycbc_fit_sngls_over_multiparam]: again say this is a list corresponding to the fit parameters

File: bin/hdfcoinc/pycbc_fit_sngls_over_multiparam [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+                         "logs of them) to smooth over. Required. For log "
+                         "template duration, try 0.2")
+args = parser.parse_args()
+pycbc.init_logging(args.verbose)
+
```
Comment (tdent) [inline bin/hdfcoinc/pycbc_fit_sngls_over_multiparam]: Add input checking here: `assert len(args.log_param) == len(args.fit_param)` and likewise for `args.smoothing_width`

Comment (tdent) [inline bin/hdfcoinc/pycbc_fit_sngls_over_multiparam]: Might also be nice to check immediately that the `args.log_param` values are valid before spending time calculating parameters, however consider this as optional

File: bin/hdfcoinc/pycbc_fit_sngls_over_multiparam [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+alpha_smoothed = []
+rang = numpy.arange(0, len(nabove))
+for i in range(len(nabove)):
+    dsq = dist(i, rang)
+    l = (dsq < 1)
+    
+    nabove_smoothed.append(nabove[l].mean())
+    alpha_smoothed.append(nabove_smoothed[i] / invalphan[l].mean())
+    
+    print i, l.sum(), m1[i], m2[i], alpha_smoothed[-1], nabove_smoothed[-1]
```
Comment (tdent) [inline bin/hdfcoinc/pycbc_fit_sngls_over_multiparam]: remove the print or convert it into logging !

**Comment by tdent** [comment]
Dammit .. github lost my comments.  Well, if you've addressed them all and travis is happy then consider it approved.

**Comment by ahnitz** [comment]
@tdent Yes, I've addressed the ones that have closed.

**Pull request merged**